paymentintent/client.go

```diff
@@ -0,0 +1,129 @@
+	err := c.B.Call(http.MethodPost, path, c.Key, params, payment_intent)
+	return payment_intent, err
+}
+
+// Cancel cancels a pending payment_intent.
+// For more details see https://stripe.com/docs/api#cancel_payment_intent.
+func Cancel(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	return getC().Cancel(id, params)
+}
+
+func (c Client) Cancel(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	path := stripe.FormatURLPath("/payment_intents/%s/cancel", id)
+	payment_intent := &stripe.PaymentIntent{}
+	err := c.B.Call(http.MethodPost, path, c.Key, params, payment_intent)
+	return payment_intent, err
+}
+
+// Capture captures a pending payment_intent.
+// For more details see https://stripe.com/docs/api#capture_payment_intent.
+func Capture(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	return getC().Capture(id, params)
+}
+
+func (c Client) Capture(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	path := stripe.FormatURLPath("/payment_intents/%s/capture", id)
+	payment_intent := &stripe.PaymentIntent{}
+	err := c.B.Call(http.MethodPost, path, c.Key, params, payment_intent)
+	return payment_intent, err
+}
+
+// Confirm confirms a pending payment_intent.
+// For more details see https://stripe.com/docs/api#confirm_payment_intent.
+func Confirm(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	return getC().Confirm(id, params)
+}
+
+func (c Client) Confirm(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	path := stripe.FormatURLPath("/payment_intents/%s/confirm", id)
+	payment_intent := &stripe.PaymentIntent{}
+	err := c.B.Call(http.MethodPost, path, c.Key, params, payment_intent)
+	return payment_intent, err
+}
+
+// List returns a list of payment_intents.
+// For more details see https://stripe.com/docs/api#list_payment_intents.
+func List(params *stripe.PaymentIntentListParams) *Iter {
+	return getC().List(params)
+}
+
+func (c Client) List(listParams *stripe.PaymentIntentListParams) *Iter {
+	return &Iter{stripe.GetIter(listParams, func(p *stripe.Params, b *form.Values) ([]interface{}, stripe.ListMeta, error) {
+		list := &stripe.PaymentIntentList{}
+		err := c.B.CallRaw(http.MethodGet, "/payment_intents", c.Key, b, p, list)
+
+		ret := make([]interface{}, len(list.Data))
+		for i, v := range list.Data {
+			ret[i] = v
+		}
+
+		return ret, list.ListMeta, err
+	})}
+}
+
+// Iter is an iterator for lists of PaymentIntents.
+// The embedded Iter carries methods with it;
+// see its documentation for details.
+type Iter struct {
+	*stripe.Iter
+}
+
+// PaymentIntent returns the most recent PaymentIntent
+// visited by a call to Next.
+func (i *Iter) PaymentIntent() *stripe.PaymentIntent {
+	return i.Current().(*stripe.PaymentIntent)
+}
+
+func getC() Client {
+	return Client{stripe.GetBackend(stripe.APIBackend), stripe.Key}
+}
```

Comment:
Would you mind taking a stab at getting the new `paymentintent/` package to pass Golint? Basically you'd be doing what I did [in the PR that activated it for account and charge](https://github.com/stripe/stripe-go/pull/593/files):

1. Get Golint running locally:

    ```
    go get -u golang.org/x/lint/golint
    golint ./paymentintent
    ```

2. Fix the linting problems (using the same style of comment from `./charge` basically:

    ```
    paymentintent/client.go:1:1: package comment should be of the form "Package paymentintent ..."
    paymentintent/client.go:23:1: exported method Client.New should have comment or be unexported
    paymentintent/client.go:24:2: don't use underscores in Go names; var payment_intent should be paymentIntent
    paymentintent/client.go:35:1: exported method Client.Get should have comment or be unexported
    paymentintent/client.go:37:2: don't use underscores in Go names; var payment_intent should be paymentIntent
    paymentintent/client.go:48:1: exported method Client.Update should have comment or be unexported
    paymentintent/client.go:50:2: don't use underscores in Go names; var payment_intent should be paymentIntent
    paymentintent/client.go:61:1: exported method Client.Cancel should have comment or be unexported
    paymentintent/client.go:63:2: don't use underscores in Go names; var payment_intent should be paymentIntent
    paymentintent/client.go:74:1: exported method Client.Capture should have comment or be unexported
    paymentintent/client.go:76:2: don't use underscores in Go names; var payment_intent should be paymentIntent
    paymentintent/client.go:87:1: exported method Client.Confirm should have comment or be unexported
    paymentintent/client.go:89:2: don't use underscores in Go names; var payment_intent should be paymentIntent
    paymentintent/client.go:100:1: exported method Client.List should have comment or be unexported
    ```

3. Add the package to the whitelist in `Makefile`:

	```
	lint:
		golint -set_exit_status ./account
		golint -set_exit_status ./charge
		golint -set_exit_status ./paymentintent
	```